crates/rpc-types-eth/src/filter.rs

```diff
@@ -213,34 +222,44 @@ impl FilterBlockOption {
     }
 
     /// Sets the block number this range filter should start at.
-    #[must_use]
-    pub fn with_from_block(&self, block: BlockNumberOrTag) -> Self {
+    pub fn with_from_block(&self, block: BlockNumberOrTag) -> Result<Self, FilterBlockError> {
```

Comment:
this is very inconvenient, we should not do this.

I'm fine with an additional ensure_ function that does this check, but this should not be part of the builder functions because not useful

Reply:
Just fixed that, let me know :)

I tried, apparently you didn't like it haha

In fact, it was while testing our RPC that I realized that it seemed much too easy to produce erroneous block ranges and then spend several minutes before understanding what was wrong.

For example in our case we filter with the MongoDB Rust client and therefore obviously if I filter the blocks from 15 to 2 it doesn't work but I had to do a lot of prints before understanding that I was doing that :)

So I thought that an unchecked method (for the more confident) and a checked method (for the more cautious) could be a good idea. But let's do this with an `ensure_` method that at least allows us to have the option of adding risk insurance.